src/backend/distributed/planner/distributed_planner.c

```diff
@@ -835,16 +838,9 @@ GetDistributedPlan(CustomScan *customScan)
 	node = (Node *) linitial(customScan->custom_private);
 	Assert(CitusIsA(node, DistributedPlan));
 
-	node = CheckNodeCopyAndSerialization(node);
+	CheckNodeCopyAndSerialization(node);
 
-	/*
-	 * When using prepared statements the same plan gets reused across
-	 * multiple statements and transactions. We make several modifications
-	 * to the DistributedPlan during execution such as assigning task placements
-	 * and evaluating functions and parameters. These changes should not
-	 * persist, so we always work on a copy.
-	 */
-	distributedPlan = (DistributedPlan *) copyObject(node);
+	distributedPlan = (DistributedPlan *) node;
```

Comment:
We are now putting burden of copying this object to callers. Are we sure we want to do this ?

Function description mentions this to new callers. My worry is that, when somebody makes change in one of the callers to modify the distributed plan we may have problems.

Reply:
True, but this optimisation is worth getting. What if we name the function something like `GetDistributedPlanPointer` to make it clear that it does not make a copy?

Reply:
Current name is good as it stands I think. I did not like adding  ```Pointer``` suffix much.